File: awa/sdk_dist/python/awa/client/utils/workflow/build_prompt_workflow.py
from pathlib import Path
from typing import Any

from temporalio import workflow

from awa.client import constants
from awa.client.models.input_params import InputParams


async def build_prompt_workflow(
    template_input: InputParams,
    variables: dict[str, Any] | None = None,
    inputs: list[InputParams] | None = None,
    output_path: str | Path | None = None,
) -> str:
    """Build a prompt using a template and variables.

    Args:
        template_input: InputParams containing template configuration and content.
        variables: Optional dictionary of variables to use in template resolution.
            Defaults to None.
        inputs: Optional list of InputParams for the prompt building process.
            Defaults to None.
        output_path: Optional path where the built prompt should be saved.
            Can be a string or Path object. Defaults to None.

    Returns:
        str: The built prompt string.

    """
    return await workflow.execute_child_workflow(
        workflow=constants.WORKFLOW_BUILD_PROMPT,
        arg={
            "template_input": template_input,
            "variables": variables,
            "inputs": inputs,
            "output_path": str(output_path) if output_path else None,
        },
        task_queue=constants.AWA_DEFAULT_TASK_QUEUE,
    )

File: awa/sdk_dist/csharp/Awa.Client/Utilities/Workflow/BuildPromptWorkflow.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Awa.Client.Constants;
using Awa.Client.Models;

namespace Awa.Client.Utilities.Workflow
{
    public static partial class WorkflowUtilities
    {
        /// <summary>
        /// Build a prompt using a template and variables.
        /// </summary>
        /// <param name="templateInput">InputParams containing template configuration and content.</param>
        /// <param name="variables">Optional dictionary of variables to use in template resolution. Defaults to null.</param>
        /// <param name="inputs">Optional list of InputParams for the prompt building process. Defaults to null.</param>
        /// <param name="outputPath">Optional path where the built prompt should be saved. Defaults to null.</param>
        /// <returns>The built prompt string.</returns>
        public static async Task<string?> BuildPromptWorkflow(
            InputParams templateInput,
            Dictionary<string, object?>? variables = null,
            InputParams[]? inputs = null,
            string? outputPath = null)
        {
            var args = new object?[]
            {
                new
                {
                    template_input = templateInput,
                    variables = variables,
                    inputs = inputs,
                    output_path = outputPath
                }
            };

            return await Temporalio.Workflows.Workflow.ExecuteChildWorkflowAsync<string?>(
                AwaConstants.WorkflowBuildPrompt,
                args,
                new()
                {
                    TaskQueue = AwaConstants.AwaDefaultTaskQueue
                });
        }
    }
}
